libs/fluxc-plugin/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/GatewaysDao.kt

```diff
@@ -0,0 +1,49 @@
+package org.wordpress.android.fluxc.persistence.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Upsert
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.LocalId
+import org.wordpress.android.fluxc.persistence.entity.GatewayEntity
+
+@Dao
+abstract class GatewaysDao {
+    @Query(
+        """
+        SELECT * FROM GatewayEntity
+        WHERE siteId = :siteId
+        """
+    )
+    abstract suspend fun getGatewaysForSite(siteId: LocalId): List<GatewayEntity>
+
+    @Query(
+        """
+        SELECT * FROM GatewayEntity
+        WHERE siteId = :siteId
+        AND gatewayId = :gatewayId
+        LIMIT 1
+        """
+    )
```

Comment:
```suggestion
    @Query(
        """
        SELECT * FROM GatewayEntity
        WHERE siteId = :siteId
        AND gatewayId = :gatewayId
        """
    )
```

`siteId` and `gatewayId` are already guaranteed to be unique, so this query will always return 1 result if there's any match so `LIMIT 1` doesn't provide any benefits - I suggest we remove it.

Reply:
FYI: This is now done: dc01601f92a741f5e251454ceef07cb494601794 @wzieba